evalml/model_understanding/prediction_explanations/_algorithms.py

```diff
@@ -69,11 +67,11 @@ def _compute_shap_values(pipeline, features, training_data=None):
         if ws:
             logger.debug(f"_compute_shap_values TreeExplainer: {ws[0].message}")
         shap_values = explainer.shap_values(features, check_additivity=False)
-        # shap only outputs values for positive class for Catboost binary estimators.
+        # shap only outputs values for positive class for Catboost/Xgboost binary estimators.
         # this modifies the output to match the output format of other binary estimators.
         # Ok to fill values of negative class with zeros since the negative class will get dropped
         # in the UI anyways.
-        if estimator.model_family == ModelFamily.CATBOOST and pipeline.problem_type == ProblemTypes.BINARY:
+        if estimator.model_family in {ModelFamily.CATBOOST, ModelFamily.XGBOOST} and pipeline.problem_type == ProblemTypes.BINARY:
```

Comment:
I see, so xgboost's output format is the same as catboost's output format, and therefore the same reshaping is needed here. Sounds good. I think its strange that shap has this behavior!